mountpoint-s3/src/inode/readdir.rs

```diff
@@ -67,15 +67,13 @@ impl ReaddirHandle {
     ) -> Result<Self, InodeError> {
         let local_entries = {
             let inode = inner.get(dir_ino)?;
-            let kind_data = &inode.inner.sync.read().unwrap().kind_data;
+            let inode_state = &inode.get_inode_state()?;
```

Comment:
inline. Note that this is the state of the parent. The ones on line 76 are for the children!